src/Compiler/Compiler.php

```diff
@@ -182,9 +182,12 @@ private function compileDefinition(string $entryName, Definition $definition) :
                 }
                 break;
             case $definition instanceof StringDefinition:
-                $entryName = $this->compileValue($definition->getName());
-                $expression = $this->compileValue($definition->getExpression());
-                $code = 'return \DI\Definition\StringDefinition::resolveExpression(' . $entryName . ', ' . $expression . ', $this->delegateContainer);';
+                $expression = $definition->getExpression();
+                $callback = function (array $matches) use ($definition) {
+                    return '\'.$this->resolveStringPlaceholder(' . $this->compileValue($matches[1]) . ', ' . $this->compileValue($definition->getName()) . ').\'';
+                };
+                $value = preg_replace_callback('#\{([^\{\}]+)\}#', $callback, $expression);
```

Comment:
I would move REGEXP to class const, sth like `const STRING_VALUE_REGEX = '...'`

Reply:
yes, extracting the REGEXP to a single place is a good idea, since the same pattern is also used in https://github.com/ovos/PHP-DI/blob/feature/compiler-optimize-string-definitions/src/Definition/StringDefinition.php#L90